block.go

```diff
@@ -0,0 +1,53 @@
+package sdk
+
+import (
+	"context"
+
+	ctypes "github.com/cometbft/cometbft/rpc/core/types"
+	cosmos "github.com/cosmos/cosmos-sdk/client"
+)
+
```

Comment:
Add an enforcement that the appropriate structure implements the expected interface

Reply:
This is done through adding a new `block_test.go` which covers `BlockClient` method(s) and also serves as an example of how to use it. This is done intentionally to keep interfaces on the consumer side. Any changes to `BlockClient` method(s) will need to be accompanied by a semver version update (major/minor/patch/etc.) to avoid breaking any dependent applications, to enforce the set of signatures for current methods.